feat: respect cancelRefetch flag for infinite queries (fetchNextPage, fetchPreviousPage)

lgenzelis opened the pull request:
See issue #2617

TkDodo reviewed (commented):
awesome. could you please
- update the api docs (https://react-query.tanstack.com/reference/useInfiniteQuery), as `options.pageParam` is also mentioned for `fetchNextPage` / `fetchPreviousPage`)
- add a test for passing `cancelRefetch: false`

lgenzelis commented:
@TkDodo I'm sorry, I feel a bit lost about the testing part. Can you point me to an example where `cancelRefetch` is being tested? (for regular non-infinite queries, I mean)

TkDodo commented:
I would probably just test if `cancel` on the `query` itself has been called:

https://github.com/tannerlinsley/react-query/blob/e6da990703482cff8ea99f5a254975cb57463ed5/src/core/query.ts#L362-L364

lgenzelis commented:
@TkDodo I addressed your points 😃 
- updated documentation
- added tests for the cases where `cancelRefetch` is omitted (so, it's `true`) and when `cancelRefetch: false` is passed to `fetchNextPage`

TkDodo commented:
Would you consider this a feature rather than a bugfix ?

tannerlinsley commented:
:tada: This PR is included in version 3.26.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/react-query/v/3.26.0)
- [GitHub release](https://github.com/tannerlinsley/react-query/releases/tag/v3.26.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

lgenzelis commented:
> Would you consider this a feature rather than a bugfix ?

@TkDodo I guess so. Mainly because we didn't agree that the previous behavior was a bug on the first place 🤷‍♀️
